main.c

```diff
@@ -0,0 +1,229 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define FILENAME "users.txt"
+
+struct User {
+    int id;
+    char name[50];
+    int age;
+};
+
+// new user
+void createUser() {
+    struct User user;
+
+    // check for unique id
+    FILE *file = fopen(FILENAME, "r");
+    if (!file) {
+        printf("Error opening file!\n");
+        return;
+    }
+    int id ;
+
+    printf("Enter User ID: ");
+    scanf("%d", &id);
+    getchar();

```

Comment:
Why did you use `getchar()` here?

Reply:
is used to consume the the newline character \n  left in the input buffer.